### PR TITLE
Custom User Example

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
Extend our example to show case custom user data

This closes # ??? 

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by elle-j — COMMENTED** [review]
Just some initial comments, will look more after Skunkworks 👍

**Review by kneth — APPROVED** [review]

File: examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx [hunk 1]
```diff
@@ -121,6 +125,13 @@ export function StoreScreen() {
                 }}
                 text="Delete User"
               />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={() => {
+                  user.refreshCustomData();
+                }}
+                text="Refresh User Data"
+              />
```
[the remark this answers is not in the document]
Comment (takameyer) [reply examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx:134]: Good point.  I wonder which one should be more prevalent.  For some, the custom data is not an issue, and for others, refresh token logic is more important.  I guess having both and having them be redundant doesn't hurt anything.
@otso Do you have an opinion?

[the remark this answers is not in the document]
Comment (elle-j) [reply examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx:134]: Yeah, perhaps just the "Refresh Access Token" button is enough. Could seem unnecessary to have two buttons doing the same thing 😛

**Review by elle-j — COMMENTED** [review]
Some updates [here](https://github.com/realm/realm-js/pull/6221) 🙂

**Review by elle-j — APPROVED** [review]
Thanks for extending the example! Including custom user data will come in handy 🙂 

(The only adjustment I think I'd make is to include only one of the buttons for refreshing the custom data.)

File: examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (takameyer) [reply examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx:143]: image
they are now combined.  Doesn't look too bad and we don't lose the language

**Pull request merged**